src/electionguard/publish.py

```diff
@@ -0,0 +1,78 @@
+from typing import List
+from jsons import set_serializer
+from os import mkdir, path
+from jsons import set_serializer, dump
+
+
+from .ballot_box import CiphertextBallot
+from .decryption_mediator import PlaintextTally
+from .election import CiphertextElectionContext, ElectionConstants, ElectionDescription
+from .key_ceremony import CoefficientValidationSet
+from .tally import CiphertextTally
+from .group import ElementModP, ElementModQ
+
+RECORD_DIR = "output" + path.sep
+COEFFICIENTS_DIR = RECORD_DIR + "coefficients" + path.sep
+BALLOTS_DIR = RECORD_DIR + "encrypted_ballots" + path.sep
+SPOILED_DIR = RECORD_DIR + "spoiled_ballots" + path.sep
+
+DESCRIPTION_FILE_NAME = "description"
+CONTEXT_FILE_NAME = "context"
+CONSTANTS_FILE_NAME = "constants"
+ENCRYPTED_TALLY_FILE_NAME = "encrypted_tally"
+TALLY_FILE_NAME = "tally"
+
+COEFFICIENT_PREFIX = "coefficient_validation_set_"
+BALLOT_PREFIX = "ballot_"
+
+
+def publish(
+    description: ElectionDescription,
+    context: CiphertextElectionContext,
+    constants: ElectionConstants,
+    coefficient_validation_sets: List[CoefficientValidationSet],
+    ciphertext_ballots: List[CiphertextBallot],
+    ciphertext_tally: CiphertextTally,
+    plaintext_tally: PlaintextTally,
+    record_directory: str = RECORD_DIR,
+) -> None:
+    """Publishes the election record as json"""
```

Comment:
These JSON files will be quite amenable to compression (gzip or otherwise), and there are a number of Python libraries that can help with this. Given the potentially quite large amount of disk space being used here, you should consider compression.

Reply:
This looks at extendability that we want to open to the users of the libraries and not confine them to. Although we can we want to minimize dependencies.

Reply:
The standard CPython distribution includes gzip support (https://docs.python.org/3/library/archiving.html). All of that bottoms out at `zlib`, which is written in C and included along with CPython. As such, we're not really adding any new dependencies. What we are doing, however, is creating more attack surface.

Ideally, we'd have our own binary format (msgpack, protobuf, etc.), but I imagine that's not on the short-term roadmap.

Reply:
let's put this in a new issue so we can more properly consider outside the scope of this PR.  binary serialization is on the road map and i think we need to look at all of that together